MapboxDirections/MBIntersection.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+
+@objc(MBIntersection)
+public class Intersection: NSObject, NSSecureCoding {
+    
+    /**
+     Index into bearings/entry array.
+     
+     Used to calculate the bearing just before the turn. Namely, the clockwise angle from true north to the direction of travel immediately before the maneuver/passing the intersection. Bearings are given relative to the intersection. To get the bearing in the direction of driving, the bearing has to be rotated by a value of 180. The value is not supplied for depart maneuvers.
+    */
+    public let approachIndex: Int?
+    
+    /**
+     Index into the bearings/entry array.
+     
+     Used to extract the bearing just after the turn. Namely, The clockwise angle from true north to the direction of travel immediately after the maneuver/passing the intersection. The value is not supplied for arrive maneuvers.
+    */
+    public let outletIndex: Int?
+    
+    /**
+     An array of booleans, corresponding in a 1:1 relationship to the bearings.
+     
+     A value of true indicates that the respective road could be entered on a valid route. false indicates that the turn onto the respective road would violate a restriction.
+     */
+    public var entry: [Bool]
+    
+    /**
+     CLLocationCoordinate2D representing the location of the intersection
```

Comment:
Missing a period. Also no need to say the type, since the type is already declared:

> The geographic coordinates at the center of the intersection.